tests/unit/test_network_connections.py

```diff
@@ -4081,6 +4095,145 @@ def test_convert_passwd_flags_nm(self):
+                            "gateway": "198.51.100.1",
+                            "metric": 2,
+                            "table": 30400,
+                        },
+                    ],
+                },
+            }
+        ]
+        self.validator = network_lsr.argument_validator.ArgValidator_ListConnections()
+        self.rt_parsing = network_lsr.argument_validator.RouteTableParsing()
+        # the connection index is 0 because there is only one connection profile
+        # defined here
+        self.connection_index = 0
+
+    def test_valid_numeric_route_tables(self):
+        """
+        Test that the value between 1 and 4294967295 are the valid value for numeric
+        route tables and the value will not be normalized
+        """
+
+        self.validator.validate_route_tables(
+            self.validator.validate(self.test_connections)[0],
+            self.connection_index,
+        )
+        self.assertEqual(
+            self.test_connections[0]["ip"]["route"][0]["table"],
+            30400,
+        )
+
+        self.test_connections[0]["ip"]["route"][0]["table"] = 200
+        self.validator.validate_route_tables(
+            self.validator.validate(self.test_connections)[0],
+            self.connection_index,
+        )
+        self.assertEqual(
+            self.test_connections[0]["ip"]["route"][0]["table"],
+            200,
+        )
+
+    def test_invalid_numeric_route_tables(self):
+        """
+        Test that the value less than 1 or greater than 4294967295 are the invalid
+        value for numeric route tables
+        """
+
+        self.test_connections[0]["ip"]["route"][0]["table"] = 0
+        val_min = 1
+        val_max = 0xFFFFFFFF
+        self.assertRaisesRegex(
+            ValidationError,
+            "route table value is %s but cannot be less then %s"
+            % (
+                self.test_connections[0]["ip"]["route"][0]["table"],
+                val_min,
+            ),
+            self.validator.validate,
+            self.test_connections,
+        )
+
+        self.test_connections[0]["ip"]["route"][0]["table"] = 4294967296
+        self.assertRaisesRegex(
+            ValidationError,
+            "route table value is %s but cannot be greater then %s"
+            % (
+                self.test_connections[0]["ip"]["route"][0]["table"],
+                val_max,
+            ),
+            self.validator.validate,
+            self.test_connections,
+        )
+
+    def test_invalid_named_route_tables(self):
+        """
+        Test that the validation error will be raised when the named route table is not
+        found in the `/etc/iproute2/rt_tables`
+        """
+
+        self.test_connections[0]["ip"]["route"][0]["table"] = "eth"
+
+        self.assertRaisesRegex(
+            ValidationError,
+            "can not find route table '%s' in `/etc/iproute2/rt_tables`"
+            % (self.test_connections[0]["ip"]["route"][0]["table"]),
```

Comment:
this unit test tests some files on the system where it's tested. You cannot do this. The unit test must work everywhere, also on a system where the user edited ther rt_tables file.

What you need to do, is actually stub/patch the code `RouteTableParsing.get_route_tables_mapping = lamda: return { ... }` -- which is the main reason to have the parsing code in a separate class, that you can stub for testing.

Reply:
I stubbed for the test cases.